code/modules/reagents/reagent_containers/glass/bottle.dm

```diff
@@ -54,6 +54,44 @@
 	..()
 	update_icon()
 
+/obj/item/weapon/reagent_containers/glass/bottle/attack_self(mob/user as mob)
+	if(!is_open_container())
+		to_chat(user, "<span class='warning'>You can't, \the [src] is closed.</span>")//Added this here and elsewhere to prevent drinking, etc. from closed drink containers. - Hinaichigo
+
+		return 0
+
+	else if(!src.reagents.total_volume || !src)
+		to_chat(user, "<span class='warning'>\The [src] is empty.<span>")
+		return 0
+
+	else
+		imbibe(user)
+		return 0
+
+/obj/item/weapon/reagent_containers/glass/bottle/proc/imbibe(mob/user) //Drink the liquid within
```

Comment:
Rather than copying this from drinks line for line, move it onto reagent_containers.